app/services/balance_transactions/creator.rb

```diff
@@ -0,0 +1,21 @@
+class BalanceTransactions::Creator
+  Params = Struct.new(:date, :comment, :amount, :user_id)
+
+  def initialize(params)
+    @params = Params.new(params[:date], params[:comment], params[:transactions_attributes][:amount],
+      params[:user_id])
+  end
```

Comment:
@marderer from fast look: this service isn't reusable (good only for controllers because of params structure). See: service should know only about those variables that he vitally needs. Service shouldn't know about `params[:transactions_attributes]`

What I suggest:
```ruby
BalanceTransactions::Creator.new(
  date: some_date,
  comment: some_comment,
  amount: some_amount,
  user_id: some_user_id
).create
```

Reply:
@marderer the same thing for `BalanceTransactions::Updater`